### PR TITLE
9011 advanced search

### PR DESCRIPTION
**Pull request by ujahio**

### CODE REVIEW

**Review by codyseibert — APPROVED** [review]

File: shared/src/persistence/dynamo/helpers/store.test.js [hunk 1]
```diff
@@ -58,3 +58,30 @@ describe('setExpiresAt', () => {
...
 });
+
+describe('getLimiterByKey', () => {
+  beforeEach(() => {
+    client.getFromDeployTable = jest
+      .fn()
+      .mockReturnValue({ maxInvocations: 5, windowTime: 1000 });
+  });
+
+  it('should set expiresAt to the value passed in and reset the id back to 1', async () => {
```
Comment (rachelschneiderman) [inline shared/src/persistence/dynamo/helpers/store.test.js]: Nit: This test description seems like a copy/pasta

File: web-api/src/app-public.js [hunk 2]
```diff
@@ -77,12 +78,12 @@ const { todaysOpinionsLambda } = require('./public-api/todaysOpinionsLambda');
 const { todaysOrdersLambda } = require('./public-api/todaysOrdersLambda');
 
 // Temporarily disabled for story 7387
```
Comment (rachelschneiderman) [inline web-api/src/app-public.js:80]: Can this comment be removed?

**Review by TomElliottFlexion — APPROVED** [review]

File: web-api/src/app-public.js [hunk 3]
```diff
@@ -95,8 +96,16 @@ app.head(
 app.get('/public-api/cases/:docketNumber', lambdaWrapper(getPublicCaseLambda));
 
 // Temporarily disabled for story 7387
```
Comment (rachelschneiderman) [inline web-api/src/app-public.js:98]: Can this comment be removed too?

File: web-api/src/app-public.js (hunk 2 above)
Comment (TomElliottFlexion) [reply web-api/src/app-public.js:80]: Not yet! The endpoint won't be turned on yet, this is just to test on dev. They'll be disabled again before deploy to mig

File: shared/src/persistence/dynamo/helpers/store.test.js (hunk 1 above)
Comment (TomElliottFlexion) [reply shared/src/persistence/dynamo/helpers/store.test.js]: It is, thanks!

File: web-api/src/middleware/advancedQueryLimiter.test.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
+const { advancedQueryLimiter } = require('./advancedQueryLimiter');
+
+let mockPersistenceGateway = {};
+
+jest.mock('../applicationContext', () => () => ({
```
Comment (rachelschneiderman) [inline web-api/src/middleware/advancedQueryLimiter.test.js:5]: Nit: Could testApplicationContext be used here instead?

File: web-api/src/middleware/advancedQueryLimiter.test.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    }));
+    res = {
+      set: jest.fn(() => ({
+        status: statusMock,
+      })),
+      status: statusMock,
+    };
+  });
+
+  it('should return a 429 response', async () => {
```
Comment (rachelschneiderman) [inline web-api/src/middleware/advancedQueryLimiter.test.js:45]: Could we add a when condition to this test name so when it fails we have a sense of why? For example "when the user has exceeded their number of searches"

File: web-api/src/middleware/advancedQueryLimiter.test.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        applicationContext: applicationContextMock,
+      },
+      res,
+      next,
+    );
+    expect(deleteKeyCountMock).toBeCalled();
+    expect(next).toBeCalled();
+  });
+
+  it('should be able to be invoked exactly 5 times before reaching next if limit is not reached', async () => {
```
Comment (rachelschneiderman) [inline web-api/src/middleware/advancedQueryLimiter.test.js:95]: Where does the number 5 come from? Is that just for test purposes?

**Pull request merged**